internal/cocoa/api_cocoa_darwin.go

```diff
@@ -38,23 +43,165 @@ var (
 	sel_instanceMethodSignatureForSelector = objc.RegisterName("instanceMethodSignatureForSelector:")
 	sel_signatureWithObjCTypes             = objc.RegisterName("signatureWithObjCTypes:")
 	sel_initWithUTF8String                 = objc.RegisterName("initWithUTF8String:")
 	sel_UTF8String                         = objc.RegisterName("UTF8String")
 	sel_length                             = objc.RegisterName("length")
+	sel_processInfo                        = objc.RegisterName("processInfo")
+	sel_isOperatingSystemAtLeastVersion    = objc.RegisterName("isOperatingSystemAtLeastVersion:")
+	sel_frame                              = objc.RegisterName("frame")
+	sel_contentView                        = objc.RegisterName("contentView")
+	sel_setBackgroundColor                 = objc.RegisterName("setBackgroundColor:")
+	sel_colorWithSRGBRedGreenBlueAlpha     = objc.RegisterName("colorWithSRGBRed:green:blue:alpha:")
+	sel_setFrameSize                       = objc.RegisterName("setFrameSize:")
+	sel_object                             = objc.RegisterName("object")
+	sel_styleMask                          = objc.RegisterName("styleMask")
+	sel_setStyleMask                       = objc.RegisterName("setStyleMask:")
+	sel_mainScreen                         = objc.RegisterName("mainScreen")
+	sel_screen                             = objc.RegisterName("screen")
+	sel_isVisible                          = objc.RegisterName("isVisible")
+	sel_deviceDescription                  = objc.RegisterName("deviceDescription")
+	sel_objectForKey                       = objc.RegisterName("objectForKey:")
+	sel_unsignedIntValue                   = objc.RegisterName("unsignedIntValue")
 )
 
-type CGFloat float64
+const NSWindowCollectionBehaviorFullScreenPrimary = 1 << 7
+
+const (
+	NSWindowStyleMaskResizable  = 1 << 3
+	NSWindowStyleMaskFullScreen = 1 << 14
+)
+
+type CGFloat = float64
 
 type CGSize struct {
 	Width, Height CGFloat
 }
 
+type CGPoint struct {
+	X, Y float64
+}
+
+type CGRect struct {
+	Origin CGPoint
+	Size   CGSize
+}
+
+type NSUInteger = uint
+type NSInteger = int
+
+type NSPoint = CGPoint
+type NSRect = CGRect
+type NSSize = CGSize
+
 type NSError struct {
 	objc.ID
 }
 
+type NSColor struct {
+	objc.ID
+}
+
+func NSColor_colorWithSRGBRedGreenBlueAlpha(red, green, blue, alpha CGFloat) (color NSColor) {
+	sig := NSMethodSignature_signatureWithObjCTypes("@@:ffff")
+	inv := NSInvocation_invocationWithMethodSignature(sig)
+	inv.SetSelector(sel_colorWithSRGBRedGreenBlueAlpha)
+	inv.SetArgumentAtIndex(unsafe.Pointer(&red), 2)
+	inv.SetArgumentAtIndex(unsafe.Pointer(&green), 3)
+	inv.SetArgumentAtIndex(unsafe.Pointer(&blue), 4)
+	inv.SetArgumentAtIndex(unsafe.Pointer(&alpha), 5)
+	inv.InvokeWithTarget(objc.ID(class_NSColor))
+	inv.GetReturnValue(unsafe.Pointer(&color))
+	return color
+}
+
+type NSOperatingSystemVersion struct {
+	Major, Minor, Patch NSInteger
+}
+
+type NSProcessInfo struct {
+	objc.ID
```

Comment:
Do we have a plan to add other members to `NS` classes? If not, would it make sence to define the type like this?

```go
type NSProcessInfo objc.ID
```

Reply:
If it's embedded then it's possible to call Send without a cast

Reply:
Ah that makes sense.